halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/Node.java

```diff
@@ -137,4 +138,15 @@ public void parentify() {
         .filter(f -> f.getDeclaredAnnotation(LocalFile.class) != null)
         .collect(Collectors.toList());
   }
+
+  public void recursiveConsume(Consumer<Node> consumer) {
+    consumer.accept(this);
+
+    NodeIterator children = getChildren();
+    Node child = children.getNext();
+    while (child != null) {
```

Comment:
Could this loop be replaced with a `children.forEachRemaining`? Or is children not an iterator?